src/audio/kpb.c

```diff
@@ -541,8 +541,12 @@ static int kpb_copy(struct comp_dev *dev)
 		 * use by clients.
 		 */
 		if (source->avail <= kpb->kpb_buffer_size) {
-			kpb_buffer_data(kpb, source, copy_bytes);
-
+			ret = kpb_buffer_data(kpb, source, copy_bytes);
+			if (ret) {
```

Comment:
It would be better to check for ret < 0 in case the function will return some non-error positive status in the future.

Reply:
@tlauda but the point is `kpb_buffer_data()` will return non zero value only if there was some error. Sometimes it will return negative linux codes, sometimes positive sof errors like `PPL_STATUS_PATH_STOP`.